modules/apps/commerce/commerce-inventory-service/src/main/java/com/liferay/commerce/inventory/internal/upgrade/v3_0_0/CommerceInventoryWarehouseUpgradeProcess.java

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.commerce.inventory.internal.upgrade.v3_0_0;
+
+import com.liferay.commerce.inventory.internal.upgrade.v3_0_0.util.CommerceInventoryWarehouseTable;
+import com.liferay.commerce.inventory.model.CommerceInventoryWarehouse;
+import com.liferay.commerce.inventory.service.CommerceInventoryWarehouseLocalService;
+import com.liferay.portal.kernel.language.LanguageUtil;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.model.Company;
+import com.liferay.portal.kernel.service.CompanyLocalService;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.kernel.util.HashMapBuilder;
+import com.liferay.portal.kernel.util.LocalizationUtil;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.Statement;
+
+import java.util.Locale;
+
+/**
+ * @author Andrea Sbarra
+ */
+public class CommerceInventoryWarehouseUpgradeProcess extends UpgradeProcess {
+
+	public CommerceInventoryWarehouseUpgradeProcess(
+		CommerceInventoryWarehouseLocalService
+			commerceInventoryWarehouseLocalService,
+		CompanyLocalService companyLocalService) {
+
+		_commerceInventoryWarehouseLocalService =
+			commerceInventoryWarehouseLocalService;
+		_companyLocalService = companyLocalService;
+	}
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		alter(
+			CommerceInventoryWarehouseTable.class,
+			new AlterColumnType("name", "STRING null"),
+			new AlterColumnType("description", "STRING null"));
+
+		if (_log.isInfoEnabled()) {
+			_log.info("Change column type to STRING");
+		}
+
+		try (PreparedStatement preparedStatement = connection.prepareStatement(
+				"update CIWarehouse set name = ?,description = ? where " +
+						"CIWarehouseId = ?");
+			Statement s = connection.createStatement();
+			ResultSet resultSet = s.executeQuery(
+				"select distinct CIWarehouseId from CIWarehouse")) {
+
+			while (resultSet.next()) {
+				long ciWarehouseId = resultSet.getLong("CIWarehouseId");
+
+				CommerceInventoryWarehouse commerceInventoryWarehouse =
+					_commerceInventoryWarehouseLocalService.
```

Comment:
I think this service is not even available, you should retrieve data directly from the SQL run result-set

Reply:
this actually fixed the UP